pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -429,6 +431,25 @@ public AtomicLong getPendingWriteOps() {
         return pendingWriteOps;
     }
 
+    public CompletableFuture<Void> unloadSubscription(String subName) {
+        final PersistentSubscription sub = subscriptions.get(subName);
+        if (sub == null) {
+            return CompletableFuture.failedFuture(new RestException(Response.Status.NOT_FOUND,
```

Comment:
I don't think throwing `RestException` in the broker layer is a good approach. RestException should only throw by the web service layer.

Reply:
> I don't think throwing RestException in the broker layer is a good approach. RestException should only throw by the web service layer.

I thought about this and decided to use RestException instead of the others because this method will only be used for Web services now, and we shouldn't over-design for a scenario that doesn't exist. It is better to redesign the exception when there is another requirement to call this method。

Reply:
>It is better to redesign the exception when there is another requirement to call this method。

Since Apache Pulsar is an open-source project, suppose you hand over this responsibility to the next contributor. The odds are that he will simply use this exception. This will create a large amount of technical debt.

Reply:
> The odds are that he will simply use this exception. This will create a large amount of technical debt.
No. Neither of these exceptions should occur because users will not try to unload a sub that does not exist, nor do they try to unload an unsupported Sub which named `__compaction`.

As mentioned above, I share your feeling that the use of an `RestException` exception here is unreasonable, and has now been replaced with a dedicated exception.